ast2vec/repo2/base.py

```diff
@@ -58,6 +59,42 @@ def __init__(self, tempdir=None, linguist=None, log_level=logging.INFO,
                         for _ in range(multiprocessing.cpu_count())]
         self._timeout = timeout
 
+        # Determines the input of convert_uasts function.
+        # True if we want to get only uast. False if uast and source filename
+        # False is used in such features extraction classes as
+        # :class:ast2vec.repo2.source.Repo2Source
+        self._uast_only = True
+
+    def clone_repository(self, url, save_dir=None):
+        """
+        Clone repository from provided url and saves it to directory target_dir.
+        if target_dir is not provided it saves to temp folder and return it name
+        :param url: a URL to clone
+        :param save_dir: path to save cloned repo
+        :return: target_dir or temp folder path
+        """
+        if save_dir is None:
+            target_dir = tempfile.mkdtemp(prefix="repo2base-", dir=self._tempdir)
+        else:
+            target_dir = save_dir
+        git_log_file = tempfile.mkstemp(prefix="gitlog-", dir=self._tempdir)[1]
```

Comment:
No temp files for git logs. Change check_call to check_output and make sure stderr is captured too. If check_output does not work, generic subprocess.Popen (not subprocess.run - it does not exist in py3.4). stderr=subprocess.PIPE or like that.